toolkit/docs/how_it_works/3_package_building.md

```diff
@@ -181,42 +171,22 @@ Once the packages are cached they are copied into `./../out/RPMs` for use in fur
 
-### Stage 5: Pkgworker
-The `pkgworker` tool is not invoked directly by the build system, instead it is invoked from a recursive `Make` call to the dynamically generated `workplan.mk` file.
+During the build `scheduler` will continually try to optimize the build graph. When the graph is first loaded, `scheduler` walks the dependencies of every package that needs to be built. If it does not find any dynamic dependencies it will create a subgraph that only contains the needed packages.
+If there are dynamic dependencies found then the graph cannot be subgraphed as the `scheduler` does not know which package in the graph, if any, will satisfy this dynamic dependency. As the `scheduler` fills in information about implicit provides during the build, it will continuously attempt to create a solvable subgraph and finish building using that.
 
-`Pkgworker` uses the `worker_chroot` (see [Chroot Worker](1_initial_prep.md#chroot_worker)) environment to build each package independently. First it creates an empty folder to build in (one for each package to build) and extracts the chroot archive into it. This preps the environment with all the toolchain packages which were made available during the prep stage (see [Toolchain](1_initial_prep.md#toolchain)). It then mounts the local RPM folder into the environment so the worker can access any build dependencies it has. Using `tdnf` and `rpmbuild` the worker installs the build dependencies from the local packages, then builds the require package. Once the build is complete the freshly build pacakge is placed into the `./../out/RPMS/` folder to be available to future workers.
+The `scheduler` tool outputs the final state of the dependency graph to `./../build/pkg_artifacts/built_graph.dot`.
 
-Because the dependency information has been encoded in `workplan.mk` it is possible to build multiple pacakges in parallel. `Make` will guarantee that no package is built before its `BuildRequires` are all satisfied as encoded in the graph.
+### Stage 4: Pkgworker
+The `pkgworker` tool is not invoked directly by the build system. Instead it is invoked from the `scheduler` tool.
+`pkgworker` uses the `worker_chroot` (see [Chroot Worker](1_initial_prep.md#chroot_worker)) environment to build each package independently. First it creates an empty folder to build in (one for each package to build) and extracts the chroot archive into it. This preps the environment with all the toolchain packages which were made available during the prep stage (see [Toolchain](1_initial_prep.md#toolchain)). It then mounts the local RPM folder into the environment so the worker can access any build dependencies it has. Using `tdnf` the worker installs the build dependencies from the local packages, then using `rpmbuild` it builds the specified package. Once the build is complete the freshly build packages are placed into the `./../out/RPMS/` folder to be available to future workers.
```

Comment:
>./.. [](start = 738, length = 4)

Suggestion: to make it easier on the reader (especially one new to Mariner), it might be worth to remind them every now and then, that all relative paths start from the `toolkit` directory. Or change the paths to start from the root of the repo. I don't know how wide spread this way of providing path is, though, so maybe it's way out of scope for the PR.

Reply:
This specific snippet regarding `./../out/RPMS/` comes from the original documentation, so I'd prefer any changes to that be done in a separate PR.